src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -1596,6 +1595,11 @@ AssignTasksToConnections(DistributedExecution *execution)
 		int placementExecutionIndex = 0;
 		int placementExecutionCount = list_length(task->taskPlacementList);
 
+		if (!TransactionConnectedToLocalhost)
+		{
+			TransactionConnectedToLocalhost = TaskAccessesLocalNode(task);
```

Comment:
I'll change this flag a bit more to support more cases such as . 
```SQL
	if (!TransactionConnectedToLocalhost)
		{
			TransactionConnectedToLocalhost = TaskAccessesLocalNode(task) && modLevel > ROW_MODIFY_READONLY;
		}
```

(I'll change the name)